src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -158,15 +158,21 @@ class DeltaAnalysis(session: SparkSession, conf: SQLConf)
     val project = query.output.zipWithIndex.map { case (attr, i) =>
       if (i < targetAttrs.length) {
         val targetAttr = targetAttrs(i)
-        val expr = (attr.dataType, targetAttr.dataType) match {
+        (attr.dataType, targetAttr.dataType) match {
           case (s, t) if s == t =>
-            attr
+            attr match {
+              case AttributeReference(name, _, _, _) if name == targetAttr.name =>
```

Comment:
Sorry I may need more context: where we convert the constant to attribute?

Reply:
Maybe the title is not proper. The problem is `normalizeQueryColumns()` will add a new `Project` node with `Alias` with all attributes. But the `deleteExpr` in `OverwriteByExpression` will use the old expression. So we get Resolved attribute(s) c#792 missing from a#793,b#794,c#795.
This RP is to skip adding Alias on an attribute with the same data type and name.

Reply:
It seems a better way is to replace old attr with new one in the `deleteExpr`.

Reply:
For this yes we can replace the `deleteExpr`, but I don't know whether we need to do the same things more in future, it's easy to miss. So I fix it in normalization.

Reply:
as long as we still add alias over attribute, I think it's more robust to replace attributes when needed.

Reply:
Ok. About how to fix it, I will follow your suggestion.